nikola/plugins/task/archive.py

```diff
@@ -150,10 +152,34 @@ def provide_context_and_uptodate(self, classification, lang, node=None):
             kw["is_feed_stale"] = (datetime.datetime.utcnow().strftime("%Y/%m/%d") != classification)
         else:
             raise Exception("Cannot interpret classification {}!".format(repr(classification)))
+
         context = {
             "title": title,
             "pagekind": [page_kind, "archive_page"],
         }
+
+        # Generate links for hierarchies
+        if kw["create_archive_navigation"]:
+            if hierarchy:
+                # Up level link makes sense only if this is not the top-level
+                # page (hierarchy is empty)
+                context["up_level"] = self.site.link('/'.join(hierarchy[:-1]), lang)
+            else:
+                context["up_level"] = None
+
+            flat_hierarchy = self.site.flat_hierarchy_per_classification['archive'][lang]
+            nodelevel = len(hierarchy)
+            flat_samelevel = [node.classification_name for node in flat_hierarchy if len(node.classification_path) == nodelevel]
+            # We need to sort it. Natsort means it’s year 10000 compatible!
+            flat_samelevel = natsort.natsorted(flat_samelevel, alg=natsort.ns.F | natsort.ns.IC)
+            idx = flat_samelevel.find(classification)
+            if idx == -1:
+                raise Exception("Cannot find classification {0} in flat hierarchy!".format(classification))
+            previdx, nextidx = idx - 1, idx + 1
+            # If the previous index is -1, or the next index is 1, the previous/next archive does not exist.
+            context["previous_archive"] = self.site.link(flat_samelevel[previdx], lang) if previdx != -1 else None
+            context["next_archive"] = self.site.link(flat_samelevel[nextidx], lang) if nextidx != -1 else None
```

Comment:
I think it would make sense to also insert the names of the archives. Links for the Atom feeds (see da2x' comments in #1639) would be nice, too.